engine/src/main/java/org/terasology/rendering/opengl/GLSLShader.java

```diff
@@ -309,7 +312,7 @@ private void dumpCode(int type, Set<ShaderProgramFeature> features, String sourc
         }
 
         // Dump all final shader sources to the log directory
-        final String strippedTitle = getUrn().toString().replace(":", "-");
+        final String strippedTitle = COLON.matcher(getUrn().toString()).replaceAll(Matcher.quoteReplacement("-"));
```

Comment:
Nah. This seems to be acting on very small string, a title used to establish a file name. You'd start to notice speed-ups loading hundreds of thousands of shaders I'd imagine. Even including modules I doubt a user will ever find herself loading more than a hundred shaders.